### PR TITLE
[Merged by Bors] - doc(Topology/Metric/Lipschitz): add a few more docstrings

### PR DESCRIPTION
**Pull request by grunweg**
---

Noted when working with that file.

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/Lipschitz.lean [hunk 1]
```diff
@@ -254,6 +255,7 @@ protected theorem prod_snd : LipschitzWith 1 (@Prod.snd α β) :=
   LipschitzWith.of_edist_le fun _ _ => le_max_right _ _
 #align lipschitz_with.prod_snd LipschitzWith.prod_snd
 
+/-- The product of Lipschitz functions is Lipschitz. -/
```
Comment (kmill) [inline Mathlib/Topology/MetricSpace/Lipschitz.lean:258]: Do you want to reword this to make it clear that it's not the functions multiplied together? I'm not sure of a good wording, but give it a little thought before merging.

Thanks!

bors d+

Comment (jcommelin) [reply Mathlib/Topology/MetricSpace/Lipschitz.lean:258]: Maybe "A function to a product type created from two Lipschitz functions is itself Lipschitz"? Not ideal, but the best I could come up with...

Comment (grunweg) [reply Mathlib/Topology/MetricSpace/Lipschitz.lean]: Reworded, taking both your suggestions into account. An improvement indeed!

**Comment by grunweg** [comment]
Thanks for the fast reviews :heart:

**Comment by grunweg** [comment]
Error looks network-related (besides, a doc-only PR in mathlib shouldn't cause build failures in the archive). Does this work?
bors retry
